### PR TITLE
doc: add troubleshooting.md docs

### PR DESCRIPTION
**Pull request by Piyush23786**
add troubleshooting.md docs

### CODE REVIEW

**Review by theletterf — CHANGES_REQUESTED** [review]

File: Troubleshooting.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+
+# Troubleshooting 
+
+## Swift Static Libraries
```
[the remark this answers is not in the document]
Comment (Piyush23786) [reply Troubleshooting.md:4]: Error during build :- Could not build Objective-C module 'SQLite3'

File: Troubleshooting.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+if Swift pod depends on an Objective-C framework, a pod file needs to enable "modular headers" for that Objective-C pod
+There are multiple ways we can use modular headers based on project requirements.
+
+1.Comment use_frameworks! and add use_modular_headers!  in pod file
+
+2.In most cases for a small project it will be enough to add use_modular_headers! instead of the removed use_frameworks! Alternatively, you can try adding:modular_headers => true after each pod declaration of a "missing" module But the bigger project might contain modules that just don't want to be static, with or without modular headers.
+Ex:- pod 'ABCD', '~> 1.0.7.5', :modular_headers => true
+
+3.use_frameworks! :linkage => :static which will generate full frameworks with statically linked libraries and module map files.
```
[the remark this answers is not in the document]
Comment (Piyush23786) [reply Troubleshooting.md:18]: For a small application, which does not use multiple pods, the first approach is recommended, but for a bigger project with multiple pods that don't want to be static with or without modular headers, we can use the second case here 
It's based on a business-to-business model.

File: Troubleshooting.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (theletterf) [reply Troubleshooting.md:4]: Thanks. I've updated my suggestion. Please feel free to apply.

**Comment by theletterf** [comment]
@Piyush23786 Will approve once the suggestions have been applied.

**Review by theletterf — APPROVED** [review]

**Review by seemk — CHANGES_REQUESTED** [review]
Really shouldn't use copy-pasted sections

**Pull request closed without merging**